app/models/hardware.rb

```diff
@@ -31,7 +31,7 @@ class Hardware < ApplicationRecord
   virtual_aggregate :allocated_disk_storage, :disks, :sum, :size
 
   def ipaddresses
-    @ipaddresses ||= networks.collect(&:ipaddress).compact.uniq + networks.collect(&:ipv6address).compact.uniq
+    @ipaddresses ||= networks.pluck(:ipaddress, :ipv6address).flatten.tap(&:compact!).uniq!
```

Comment:
for some reason I'd like to see a `networks.loaded?` around this.
Wonder if the solution for our problem is just to modify the yml file and preload the networks.
(download everything solutions :( )

Reply:
Nah, I think that isn't a bad idea.  It also seems like I have broken something with this in the test suite, so I might not include this change in the final PR.  I mostly included it because we did try it when testing some things out on Friday, but probably doesn't help with the long term fix.